### ISSUE TITLE
Dynamic routing not working

### ISSUE THREAD

**Issue by imfaisalkh** [influitive/apartment#351]
I've created a multi-tenant app using `apartment` and `devise` gem. Since, I'm using postgresql, `apartment` gem will create separate schema for each tenant. My database as a few tables, `user` table is in public namespace whereas `pages` table is in each tenant specific schema. I'm able to login to each tenant separately and can create separate records for pages table of each tenant.

Now, I want these pages to be accessible in a format like this `tenant1.example.com/page-slug-here` for this I've following rule in my `routes.rb` file:

    Page.where.not(slug: nil).all.each do |page|
      get "/#{page.slug}", controller: "pages", action: "show", id: page.id
    end

These routing rules were working fine when the app was single-tenant but in multi-tenancy it's not working and is throwing "no route matches" error.

My guess here is that the database call in above routing rule is looking in `public` schema instead of tenant specific schema. How to fix this issue?

**Comment by mikecmpbll** [comment]
the routes file is processed once when your app is loaded, so the code you've posted probably isn't doing what you want it to.

use a dynamic segment or glob in the route and do the page lookup in your show action. you can probably even use a resourceful route:

```ruby
resources :pages, path: ''
```

**Comment by imfaisalkh** [comment]
You are right, the issue has been solved. Apparently, the `Page.where()...` call was being mapped to public tenant instead of requested tenant at the point of resolving of routes so I'm now using the dynamic segment in routing rule and in controller I'm using the code similiar to yours mentioned code to find the appropriate page.

I've simplified my routing rule as follows and now it is working.

`get '/:id' => 'pages#show'`

The issue can now be labelled as resolved and thanks very much for your help.
